### DISCUSSION TITLE
Cyclable lithium capacity is greater than electrode capacity when varying the electrode thickness

### DISCUSSION THREAD

**Discussion by tGisbertz** [pybamm-team/PyBaMM#2772] [Q&A] [answered]
Hello all,

I'm playing with pybamm and an optimization wrt energy density by varying the electrode thicknesses. For some values of the electrode thicknesses however I get the this warning from the electrode_soh script: warnings.warn(f"Q_Li={Q_Li:.4f} Ah is greater than Q_p={Q_p:.4f} Ah.") (e.g. when the positive electrode thickness is lower than in the original parameter set: UserWarning: Q_Li=8.4028 Ah is greater than Q_p=7.9900 Ah). 
My understanding is, that at least Qn and Qp have to be given for the esoh calculation, but but i dont really understand/find how/where those values are calculated since they are not given in the parameter set. 
What would be the best way to avoid having QLi outside of Qn and Qp when varying the thicknesses and / how can i calculate QLi?

Thanks in advance!

**Comment by valentinsulzer** [comment, marked as answer]
The relevant calculations for this are in https://github.com/pybamm-team/PyBaMM/blob/c63671db529d5e4c9e4acd965ac9b28e2bf2d272/pybamm/parameters/lithium_ion_parameters.py#L489-L514

`Q_n` and `Q_p` are linearly proportional to thickness.

Then `Q_Li = Q_Li_n + Q_Li_p = Q_n * x_100 + Q_p * y_100`, where `x_100` and `y_100` are initial stoichiometries.  So you need to find the values of `x_100` and `y_100` that give the `Q_Li` you want. You can use the `ElectrodeSOHSolver` to do this.

**Comment by valentinsulzer** [comment]
Based on your discussions raised I'd be interested in talking more with you to find out how you are using PyBaMM. Can you email me valentin@ion-works.com

**Reply by tGisbertz** [reply]
Sure I'll get in touch with you, also thanks for your quick response!
